.github/workflows/security-audit.yml

```
@@ -0,0 +1,266 @@
+      uses: actions/checkout@v4
+
+    - name: Setup Node.js (for frontend and extension)
+      if: matrix.component != 'backend'
+      uses: actions/setup-node@v4
+      with:
+        node-version: '20'
+        cache: 'pnpm'
+
+    - name: Setup Python (for backend)
+      if: matrix.component == 'backend'
+      uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)